BHoM_Engine/Modify/SetPropertyValue.cs

```diff
@@ -87,24 +87,48 @@ public static object SetPropertyValue(this object obj, string propName, object v
                         value = Activator.CreateInstance(propType);
                 }
 
-                if (propType.IsEnum && value is string)
+                if (value is string)
                 {
-                    string enumName = (value as string).Split('.').Last();
-                    try
+                    if (propType.IsEnum)
                     {
-                        object enumValue = Compute.ParseEnum(propType, enumName);
-                        if (enumValue != null)
-                            value = enumValue;
+                        string enumName = (value as string).Split('.').Last();
+                        try
+                        {
+                            object enumValue = Compute.ParseEnum(propType, enumName);
+                            if (enumValue != null)
+                                value = enumValue;
+                        }
+                        catch
+                        {
+                            Compute.RecordError($"An enum of type {propType.ToText(true)} does not have a value of {enumName}");
+                        }
                     }
-                    catch
+
+                    if (typeof(IEnum).IsAssignableFrom(propType))
                     {
-                        Compute.RecordError($"An enum of type {propType.ToText(true)} does not have a value of {enumName}");
-                    }   
-                }
+                        value = CreateEnumeration(propType, value as string);
+                    }
 
-                if (value is string && typeof(IEnum).IsAssignableFrom(propType))
-                {
-                    value = CreateEnumeration(propType, value as string);
+                    if (propType == typeof(Type))
+                        value = Create.Type(value as string);
+                    else if (propType == typeof(double))
+                    {
+                        double result = 0;
+                        double.TryParse(value as string, out result);
+                        value = result;
+                    }
+                    else if (propType == typeof(int))
+                    {
+                        int result = 0;
+                        int.TryParse(value as string, out result);
+                        value = result;
+                    }
+                    else if (propType == typeof(bool))
+                    {
+                        bool result = false;
+                        bool.TryParse(value as string, out result);
+                        value = result;
+                    }
```

Comment:
This is the only real code change. The rest is just existing code all moved under a single check for string vale for simplicity.